packages/components/src/components/hds/dialog-primitive/header.hbs

```diff
@@ -0,0 +1,32 @@
+{{! @glint-nocheck: not typesafe yet }}
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: MPL-2.0
+}}
+<div class="hds-dialog-primitive__header {{@contextualClassPrefix}}__header" ...attributes>
```

Comment:
Yes makes sense to consider the use case where the `@contextualClassPrefix` is not provided.

I see three options:
- have 5 different getters, one per element with a class (`classNamesHeader()`, `classNamesIcon()`, `classNamesTitle()`, etc)
- have a single `classNames()` getter that returns an object instead of a string, with keys `header/icon/title/etc`,  (but this would be a new pattern in our codebase)
- concatenate the strings directly in the template: `{{if @contextualClassPrefix (concat @contextualClassPrefix "__header")}}`

Personally, I think the third options is the less ugly, so I'll use this one (in case you have objections let me know)

Reply:
Done in https://github.com/hashicorp/design-system/pull/2211/commits/4749baa1fdfeffb24f3c9b5f46d6437481107945

Reply:
the last option makes sense to me too